singularity-eos/eos/eos_builder.hpp

```diff
@@ -33,6 +33,8 @@ namespace EOSBuilder {
 enum class EOSType {
   IdealGas,
   Gruneisen,
+  SNLGruneisen,
+//  Vinet,
```

Comment:
So, The SNLMG is just a placeholder. It contains only the same code as in eos_gruneissen.hpp except for naming. That is why it works in EosBuilder.

I use while because I need to know when the array starts to have numbers in it (when counting from last element).

It is not the ultimate user who will fix the array size for the parameters but the developer that implements the Vinet in their code. If you have a way to send the full array and not a pointer to the Vinet constructor, that would be best. I could not find a way. In any case, the developer in the host code need to know what size array they should send in so it we cannot rely on them reading the instructions enough to find the size out we need to add a MAXVINETPARAMETERS that they can ask for. Can I check on the size of the array the pointer is pointing at? Then I could introduce a check that this is OK as well.

Reply:
I mention it in my comments (which will appear momentarily) but I think the best way to do this is with `std::vector`.